bbq/compiler/compiler.go

```diff
@@ -2486,6 +2506,10 @@ func (c *Compiler[_, _]) emitTransfer(targetType sema.Type) {
 
 	lastInstruction := c.codeGen.LastInstruction()
 
+	// TODO: Revisit the below logic: last instruction may not always be the
+	//  actually executed last instruction, in case where branching is present.
+	//  e.g: conditional-expression
```

Comment:
Good catch! Could you please comment out the whole incorrect optimization?

Reply:
I checked the remaining optimizations, and they seemed to make sense. e.g: skipping transfer on constant loading happens if the target type is same as the loaded constant. That's why I left those as-is for now.

We can revisit them again, and disable only if they are actually problematic?

Reply:
Don't the other optimizations suffer from the same problem, that a jump could have targeted the transfer, and the previous instruction is not the previously executed instruction?

Maybe I'm misunderstanding why only the nil case is problematic / commented out

Reply:
Problem with the `nil` case is that it doesn't check the expected/target type. So if the target type is a super type, say `T?`, then then the other branch (if exists) can have a different value, and need tarnsfering/conversion.

On the other cases, since the target type is checked and is optimized only if the value type and the target type is the same, then even if there is another path, the value coming from that path would also be the same.

Anyways, I'll comment out the entire block if we are having doubts. 👍

Reply:
Disabled in https://github.com/onflow/cadence/pull/3926/commits/067f37ff3813dc754bacfa45c79c2031248e440a